tools/canonicalize-proto-file/Main.hs

```diff
@@ -116,16 +122,15 @@ instance Canonicalize [DotProtoMessagePart] where
         FieldRange _ _ -> False
         ReservedIdentifier _ -> True
 
-instance CanonicalRank DotProtoMessagePart
-         (Either (Either (Int, DotProtoIdentifier) ())
-                 (Maybe FieldNumber)) where
+instance CanonicalRank DotProtoMessagePart PartRank where
   canonicalRank = \case
-    DotProtoMessageField f -> Right (canonicalRank f)
-    DotProtoMessageOneOf _ fs -> Right (canonicalRank fs)
-    DotProtoMessageDefinition d -> Left (Left (canonicalRank d))
-    DotProtoMessageReserved _fs -> Left (Right ())
-      -- We use '()' here because 'Canonicalize [DotProtoMessagePart]'
+    DotProtoMessageOption x -> PartRankOption x
+    DotProtoMessageDefinition d -> PartRankDefinition (canonicalRank d)
+    DotProtoMessageReserved _fs -> PartRankReserved
+      -- We don't use '_fs' here because 'Canonicalize [DotProtoMessagePart]'
```

Comment:
This comment needs to be relocated above the `DotProtoMessageOneOf` case after your change

Reply:
But it was like this before. Comment is related to `DotProtoMessageReserved _fs` AST node. So it depends on style of codebase - should I put comment after related to comment line, or before.

https://github.com/awakesecurity/proto3-suite/blob/c40e73eb9a10e32d033aea3e2c8335af6c4c0926/tools/canonicalize-proto-file/Main.hs#L127-L128